src/app-bundles/districts-and-basins-bundle.js

```diff
@@ -47,20 +47,22 @@ export default createRestBundle( {
       }
     ),
 
-    doDistrictsAndBasinsFormatData: (data) => {
+    doDistrictsAndBasinsFormatData: ( /** @type a2w.models.DistrictBasin[] */ data) => {
       const districts = {};
-      data.forEach(({ district_office_id, district_name, basin_name, basin_location_code, basin_location_id }) => {
+      data.forEach(({
+        district_office_id, district_name, basin_name, basin_location_id, basin_division_code,
+        basin_operating_area_id, latitude, longitude
+      }) => {
         if (!districts[district_office_id]) {
           districts[district_office_id] = {
-            district_office_id,
-            district_name,
+            district_office_id, district_name, basin_name, basin_location_id, basin_division_code,
```

Comment:
We don't need to store the basins data within the districts object, and likewise the basins object shouldn't store the districts data. The districts and basins data is structured so that each basin contains an array of its districts. So we take the original districts and basins array from the backend, something like
```
[
```